controllers/binding/snapshotenvironmentbinding_adapter.go

```diff
@@ -99,6 +104,37 @@ func (a *Adapter) EnsureIntegrationTestPipelineForScenarioExists() (reconciler.O
 	return reconciler.ContinueProcessing()
 }
 
+// EnsureEphemeralEnvironmentsCleanedUp will ensure that ephemeral environment(s) associated with the
+// SnapshotEnvironmentBinding are cleaned up.
+func (a *Adapter) EnsureEphemeralEnvironmentsCleanedUp() (reconciler.OperationResult, error) {
```

Comment:
Why do we use a reconciliation loop for this rather than initiating cleanup in one of the functions that determines that the tests have completed?  I imagine that would use fewer resources since we're not running another function over and over again.

Reply:
This loop is there for when the provisioning fails with an unforseen error (infra or similar) so in that scenario the tests would not have even started.